File: src/main/java/me/alexirving/bot/commands/Help.java
package me.alexirving.bot.commands;

import me.alexirving.bot.utils.Messages;
import me.alexirving.bot.utils.Utils;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;

public class Help {
    public void helpCommand(SlashCommandEvent event) {
        String guildId = event.getGuild().getId();
        EmbedBuilder builder = new EmbedBuilder();
        builder.setThumbnail(Utils.getConfigMessage(guildId, Messages.HELP_AVATAR_URL));
        builder.setTitle("PteroBot help");
        builder.addField("> /link <API_KEY>", "• Links your discord account to ptero account.", false);
        builder.addField("> /server (/s, /servers)", "• Server control panel.", false);
        builder.addField("> /reinstall", "• Reinstall current selected server", false);
        builder.addField("> /rename", "• Rename current selected server", false);
        builder.addField("> /send <command> ", "• Send a command to the current selected server", false);
        event.getHook().editOriginalEmbeds(builder.build()).queue();
    }
}

File: src/main/kotlin/me/alexirving/bot/command/commands/Help.kt
package me.alexirving.bot.command.commands

import me.alexirving.bot.command.Command
import me.alexirving.bot.command.CommandHandler
import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.Permission
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent

class Help(val commandHandler: CommandHandler) :
    Command("help", "List all of the commands or get help with a specific one.", Permission.MESSAGE_HISTORY) {
    override fun execute(event: SlashCommandEvent) {
        val builder = EmbedBuilder()
        builder.setTitle("Help menu:")
        for (command: Command in commandHandler.commandMap.values) {
            builder.appendDescription("\n**${command.name}** - `${command.description}`")
        }
        event.replyEmbeds(builder.build()).setEphemeral(true).queue()
    }
}